### PR TITLE
Make sure audience is set when refreshing a RPT

### LINKED ISSUE #23399: Audience is list after refreshing a RPT
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

authorization-services

### Describe the bug

The audience is lost after refreshing an RPT using its corresponding refresh token.

### Version

22.0.3

### Expected behavior

The audience is kept when refreshing an RPT.

### Actual behavior

The audience is lost after refreshing an RPT using its corresponding refresh token.

### How to Reproduce?

* Obtain an access token using Client A
* Use the access token from Client A to exchange it with a RPT using Client B as the audience
* Check the RPT has Client B as the audience
* Refresh the RPT using Client A
* Check the RPT does not have Client B as the audience

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #23399

### CODE REVIEW

File: core/src/main/java/org/keycloak/representations/AccessToken.java [hunk 1]
```diff
@@ -90,13 +90,24 @@ public static class Authorization implements Serializable {
         @JsonProperty("permissions")
         private Collection<Permission> permissions;
 
+        @JsonProperty("aud")
+        private String audience;
+
```
Comment (stianst) [inline core/src/main/java/org/keycloak/representations/AccessToken.java:95]: `aud` is already available in JsonWebToken here, and can be either a string or a string array:
https://github.com/keycloak/keycloak/blob/ceeb604f1cf73db5dbd70a3b9e4ec56606a1982b/core/src/main/java/org/keycloak/representations/JsonWebToken.java#L55-L58

File: services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java [hunk 2]
```diff
@@ -361,6 +361,7 @@ private AuthorizationResponse createAuthorizationResponse(KeycloakIdentity ident
         AccessToken rpt = responseBuilder.getAccessToken();
         Authorization authorization = new Authorization();
 
+        authorization.setAudience(targetClient.getClientId());
```
Comment (stianst) [inline services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java:364]: Should probably just use `addAudience` from `JsonWebToken`

**Comment by pedroigor** [comment]
Thanks. Closing because I can't reproduce the original issue anymore. There were reasons, which I don't remember, for making the audience available from the `authorization` claim.

**Comment by GithubUser8080** [comment]
Hi, i would like to ask if that means that in Keycloak 23+ RPTs can be refreshed successfully without your modifications

**Pull request closed without merging**